src/Umbraco.Web.BackOffice/AspNetCore/UmbracoBackOfficeServiceCollectionExtensions.cs

```diff
@@ -30,53 +42,116 @@ public static IServiceCollection AddUmbracoConfiguration(this IServiceCollection
             return services;
         }
 
-        public static IServiceCollection AddUmbracoBackOffice(this IServiceCollection services)
+
+        /// <summary>
+        ///  Adds the Umbraco Back Core requirements
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// Must be called after all services are added to the application because we are cross-wiring the container (currently)
+        /// </remarks>
+        public static IServiceCollection AddUmbracoCore(this IServiceCollection services)
+        {
+            if (!UmbracoServiceProviderFactory.IsActive)
+                throw new InvalidOperationException("Ensure to add UseUmbraco() in your Program.cs after ConfigureWebHostDefaults to enable Umbraco's service provider factory");
+
+            var umbContainer = UmbracoServiceProviderFactory.UmbracoContainer;
+
+            return services.AddUmbracoCore(umbContainer, Assembly.GetEntryAssembly());
+        }
+
+        public static IServiceCollection AddUmbracoCore(this IServiceCollection services, IRegister umbContainer, Assembly entryAssembly)
         {
             services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
 
-            var serviceProvider = services.BuildServiceProvider();
+            CreateCompositionRoot(services);
 
-            var httpContextAccessor = serviceProvider.GetService<IHttpContextAccessor>();
-            var webHostEnvironment = serviceProvider.GetService<IWebHostEnvironment>();
-            var hostApplicationLifetime = serviceProvider.GetService<IHostApplicationLifetime>();
+            // TODO: Get rid of this 'Current' requirement
+            var globalSettings = Current.Configs.Global();
+            var umbracoVersion = new UmbracoVersion(globalSettings);
 
-            var configs = serviceProvider.GetService<Configs>();
+            // TODO: Currently we are not passing in any TypeFinderConfig (with ITypeFinderSettings) which we should do, however
```

Comment:
After the changes to config, this should be possible now

Reply:
Ok, i will make this change in the next PR since I'm refactoring a little bit of this already.